### PR TITLE
[go-utils/pagination] Granularized  pagination functionality & add total page

### PR DESCRIPTION
**Pull request by yohanesmichael72**
-  Granularized pagination functionality
-  add `TotalData` & `TotalPage` functionality
-  add maximum amount for limit

### CODE REVIEW

**Review by Amnestia — APPROVED** [review]

**Comment by yohanesmichael72** [comment]
@yafetzz @agusID

File: pagination/pagination.go [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+		p.SetToDefault()
 	}
-	if p.Page < 1 {
-		p.Page = 1
+}
+
+// SetTotalPage will set TotalPage value
+func (p *Pagination) SetTotalPage() {
+	if p.Total > 0 && p.Total < p.Limit {
+		p.Limit = p.Total
```
Comment (agusID) [inline pagination/pagination.go:36]:
```suggestion
	if p.TotalData > 0 && p.TotalData < p.Limit {
		p.Limit = p.TotalData
```

File: pagination/pagination.go [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
 	}
 
-	p.Offset = p.Limit * (p.Page - 1)
+	p.TotalPage = int(math.Ceil(float64(p.Total) / float64(p.Limit)))
```
Comment (agusID) [inline pagination/pagination.go:39]:
```suggestion
	p.TotalPage = int(math.Ceil(float64(p.TotalData) / float64(p.Limit)))
```

**Review by agusID — CHANGES_REQUESTED** [review]
please change `Total` to `TotalData` to define how many data in the database.

File: pagination/pagination.go (hunk 2 above)
Comment (yohanesmichael72) [reply pagination/pagination.go]: updated

File: pagination/pagination.go (hunk 1 above)
Comment (yohanesmichael72) [reply pagination/pagination.go]: updated

**Comment by yohanesmichael72** [comment]
> please change `Total` to `TotalData` to define how many data in the database.

updated

**Review by yafetzz — APPROVED** [review]

**Review by agusID — APPROVED** [review]

**Pull request merged**